Sources/GRPC/GRPCClientConnection.swift

```diff
@@ -54,111 +54,90 @@ import NIOTLS
+    return channel.pipeline.handler(type: GRPCTLSVerificationHandler.self).flatMap {
+      $0.verification
     }
   }
 
-  /// Configure an SSL handler on the channel, if one is required.
+  /// Makes a `GRPCClientConnection` from the given channel and configuration.
   ///
-  /// - Parameters:
-  ///   - mode: TLS mode to use when creating the new handler.
-  ///   - channel: The channel on which to add the SSL handler.
-  ///   - host: The hostname of the server we're connecting to.
-  ///   - errorDelegate: The error delegate to use.
-  /// - Returns: A future which will be succeeded when the pipeline has been configured.
-  private static func configureTLS(mode tls: TLSMode, channel: Channel, host: String, errorDelegate: ClientErrorDelegate?) -> EventLoopFuture<Void> {
-    let handlerAddedPromise: EventLoopPromise<Void> = channel.eventLoop.makePromise()
-
-    do {
-      guard let sslContext = try tls.makeSSLContext() else {
-        handlerAddedPromise.succeed(())
-        return handlerAddedPromise.futureResult
-      }
-
-      let sslHandler = try NIOSSLClientHandler(context: sslContext, serverHostname: host)
-      let verificationHandler = GRPCTLSVerificationHandler(errorDelegate: errorDelegate)
-
-      channel.pipeline.addHandlers(sslHandler, verificationHandler).cascade(to: handlerAddedPromise)
-    } catch {
-      handlerAddedPromise.fail(error)
+  /// - Parameter channel: The channel to use for the connection.
+  /// - Parameter configuration: The configuration used to create the channel.
+  public class func makeGRPCClientConnection(
+    channel: Channel,
+    configuration: Configuration
+  ) -> EventLoopFuture<GRPCClientConnection> {
+    return channel.pipeline.handler(type: HTTP2StreamMultiplexer.self).map { multiplexer in
+      GRPCClientConnection(channel: channel, multiplexer: multiplexer, configuration: configuration)
     }
+  }
 
-    return handlerAddedPromise.futureResult
+  /// Starts a client connection using the given configuration.
+  ///
+  /// This involves: creating a `ClientBootstrap`, connecting to a target, verifying that the TLS
+  /// handshake was successful (if TLS was configured) and creating the `GRPCClientConnection`.
+  /// See the individual functions for more information:
+  ///  - `makeBootstrap(configuration:)`,
+  ///  - `verifyTLS(channel:)`, and
+  ///  - `makeGRPCClientConnection(channel:configuration:)`.
+  ///
+  /// - Parameter configuration: The configuration to start the connection with.
+  public class func start(_ configuration: Configuration) -> EventLoopFuture<GRPCClientConnection> {
+    return makeBootstrap(configuration: configuration)
+      .connect(to: configuration.target)
+      .flatMap { channel in
+        let tlsVerified: EventLoopFuture<Void>?
+        if configuration.tlsConfiguration != nil {
+          tlsVerified = verifyTLS(channel: channel)
+        } else {
+          tlsVerified = nil
```

Comment:
How about simply using `= channel.eventLoop.makeSucceededFuture(())` in this case, and making `tlsVerified` non-optional?

Reply:
I had the same thought but it felt weird naming it `tlsVerified` if TLS wasn't being used and I couldn't come up with a reasonable name! Happy to change it if you like, though.

Reply:
Fine as is :-)